modules/mono/editor_templates/CharacterBody3D/basic_movement.cs

```diff
@@ -0,0 +1,44 @@
+// meta-description: Classic movement for gravity games (FPS, TPS, ...)
+// meta-default: true
+
+using _BINDINGS_NAMESPACE_;
+using System;
+
+public partial class _CLASS_ : _BASE_
+{
+    public const int Speed = 8;
+    public const int JumpForce = 4;
+
+    // Get the gravity from the project settings to be synced with RigidDynamicBody nodes.
+    public float gravity = (float)ProjectSettings.GetSetting("physics/3d/default_gravity");
+
+    public override void _PhysicsProcess(float delta)
+    {
+        Vector3 motionVelocity = MotionVelocity;
+
+        // Add the gravity.
+        if (!IsOnFloor())
+            motionVelocity.y -= gravity * delta;
+
+        // Handle Jump.
+        if (Input.IsActionJustPressed("ui_accept") && IsOnFloor())
+             motionVelocity.y = JumpForce;
+
+        // Get the input direction and handle the movement/deceleration.
+        Vector2 inputDir = Input.GetVector("ui_left", "ui_right", "ui_up", "ui_down");
+        Vector3 direction = Transform.basis.Xform( new Vector3(inputDir.x, 0, inputDir.y));
```

Comment:
micro-nit: you may want to remove this space.

```suggestion
        Vector3 direction = Transform.basis.Xform(new Vector3(inputDir.x, 0, inputDir.y));
```